src/screens/ListScreen.js

```diff
@@ -17,37 +18,125 @@ const styles = StyleSheet.create({
     width: '100%',
     height: '100%',
     backgroundColor: '#eee',
-    paddingTop: 140,
+    paddingTop: 180,
   },
 });
 
-export default function ListScreen({ navigation, style, data }) {
+const storeSearchStack = async value => {
+  try {
+    const jsonValue = JSON.stringify(value);
+    await AsyncStorage.setItem('@storage_Key', jsonValue);
+  } catch (e) {}
+};
+
+const SearchCardsComp = ({
+  scroll,
+  navigation,
+  marginBottom,
+  label,
+  labelAlign,
+  addHistory,
+  searchStack,
+}) => (
+  <View>
+    <Inset>
+      <Text
+        style={{
+          fontSize: 15,
+          color: '#3b3f51',
+          marginTop: 10,
+          textAlign: labelAlign,
+        }}
+      >
+        {label}
+      </Text>
+    </Inset>
+    <View style={{ marginTop: 10, marginBottom }}>
+      {scroll.map(tree => {
+        const { uuid, name, comments } = tree;
+        return (
+          <SearchCard
+            key={uuid}
+            name={name}
+            comments={comments}
+            onPress={() => {
+              navigation.push('TreeDetails', { uuid });
+              if (addHistory) {
+                if (searchStack.length < 1) {
+                  searchStack.unshift(tree);
+                } else if (tree.name !== searchStack[0].name) {
+                  searchStack.unshift(tree);
+                }
+                if (searchStack.length > 5) {
+                  searchStack.pop();
+                }
+                storeSearchStack(searchStack);
+              }
+            }}
+          />
+        );
+      })}
+    </View>
+  </View>
+);
+
+SearchCardsComp.propTypes = {
+  scroll: shape({
+    push: func,
+  }),
+  navigation: shape({
+    push: func,
+  }),
+  marginBottom: number,
+  label: string,
+  labelAlign: string,
+  addHistory: bool,
+  // eslint-disable-next-line react/forbid-prop-types
+  searchStack: array,
+};
+
+export default function ListScreen({
+  navigation,
+  style,
+  data,
+  searchStack,
+  searchQuery,
+  searchEntered,
+}) {
+  let text;
+  if (searchQuery.length === 0) {
+    text = 'All trees';
+  } else if (searchEntered === true) {
+    text = `${data.length} ${data.length === 1 ? 'result' : 'results'}`;
+  } else if (data.length === 0) {
+    text = "Oops, that's an invalid search. Try again!";
```

Comment:
data.length == 0 means nothing matches right? If so, maybe change text to specify that ("No matches found for your query"), since invalid is a little confusing here.